sklearn/_build_utils/deprecated_modules.py

```diff
@@ -270,9 +270,22 @@
 
 _FILE_CONTENT_TEMPLATE = """
 # THIS FILE WAS AUTOMATICALLY GENERATED BY deprecated_modules.py
+from . import {new_module_name}
+
+if {new_module_name}.__file__.endswith(".py"):
```

Comment:
what about cython files

Reply:
It turns out that only `_ball_tree.pyx` and `_kd_tree.pyx` defines `__all__`. Files such as `_tree.pyx` that does not define `__all__` the `from from sklearn.tree.tree import *` actually imports everything, such as `Tree`.

In short, even without this PR, importing from the compiled modules works, such as `from sklearn.tree.tree import Tree`.

Reply:
what about _ball_tree.pyx and _kd_tree.pyx then?

Reply:
All the other functions inside those files where `cdef` so they were not able to be imported in the first place.

Reply:
Well, cython files could. But I guess we don't care.